File: 002_RNA/002_RNA.py
'''
My solution to Rosalind Bioinformatics Problem 002

Title: Transcribing DNA into RNA
Rosalind ID: RNA
Rosalind #: 002
URL: http://rosalind.info/problems/rna
'''

from Bio.Seq import Seq
import Bio.Alphabet

# open txt file with sequence to transcribe
f = open('data/rosalind_rna.txt', 'r')
# read all lines by removing newline character
data = f.read().replace('\n', '')
# close file
f.close()

# assign sequence as a DNA sequence
t = Seq(data, Bio.Alphabet.IUPAC.unambiguous_dna)
# use transcribe function
o = open('output/002_RNA.txt', 'w')
o.write(str(t.transcribe()))
o.close()

File: 002_RNA/002_RNA.R
if(FALSE){
  '''
  My solution to Rosalind Bioinformatics Problem 002
  
  Title: Transcribing DNA into RNA
  Rosalind ID: RNA
  Rosalind #: 002
  URL: http://rosalind.info/problems/rna

  Goal to transcribe the DNA sequence to RNA sequence.

  Note - As with my python solutions, I prefer to implement
  methods provided by some Biology specific packages.  In 
  the case of R, Bioconductor.  For this specific instance,
  Biostrings is the package for working with sequences.
  '''
}

require(Biostrings)

file_name = "data/rosalind_rna.txt"
raw_data = readChar(file_name, file.info(file_name)$size)

# remove newline character
# Note - \r\n is a consequence of Windows files
raw_seq = gsub("\r\n","",raw_data)
dna = DNAString(raw_seq)

# transcribe DNA string to RNA
rna = as.character(RNAString(dna))

outhandle = "output/002_RNA.txt"
# nchar - size of character string
writeChar(rna, outhandle, nchar(rna))
